doc/man1/openssl-cmp.pod.in

```diff
@@ -0,0 +1,1142 @@
+=item * appendix D.4 shows PKIConf having protection
+
+=back
+
+=item B<-extracertsout> I<filename>
+
+The file where to save any extra certificates received in the extraCerts field
+of response messages.
+
+=item B<-cacertsout> I<filename>
+
+The file where to save any CA certificates received in the caPubs field of
+initializiation response (ip) messages.
+
+=back
+
+
+=head2 Client authentication options
+
+=over 4
+
+=item B<-ref> I<value>
+
+Reference number/string/value to use as fallback senderKID; this is required
+if no sender name can be determined from the B<-cert> or <-subject> options and
+is typically used when authenticating with pre-shared key (password-based MAC).
+
+=item B<-secret> I<arg>
+
+Source of secret value to use for creating PBM-based protection of outgoing
+messages and for verifying any PBM-based protection of incoming messages.
+PBM stands for Password-Based Message Authentication Code.
+This takes precedence over the B<-cert> option.
+
+For more information about the format of B<arg> see the
+B<PASS PHRASE ARGUMENTS> section in L<openssl(1)>.
+
+=item B<-cert> I<filename>
+
+The client's current certificate.
+Requires for the corresponding key to be given with B<-key>.
+The subject of this certificate will be used as the "sender" field
+of outgoing CMP messages, while B<-subjectName> may provide a fallback value.
+When using signature-based message protection, this "protection certificate"
+will be included first in the extraCerts field of outgoing messages.
+For IR this can be used for authenticating a request message
+using an external entity certificate as defined in appendix E.7 of RFC 4210.
+For KUR this is also used as certificate to be updated if the B<-oldcert>
+option is not given.
+If the file includes further certs, they are appended to the untrusted certs.
+These may get added to the extraCerts field sent in requests as far as needed.
+
+=item B<-key> I<filename>
+
+The corresponding private key file for the client's current certificate given in
+the B<-cert> option.
+This will be used for signature-based message protection unless
+the B<-secret> option indicating PBM or B<-unprotected_requests> is given.
+
+=item B<-keypass> I<arg>
+
+Pass phrase source for the private key given with the B<-key> option.
+Also used for B<-cert> and B<-oldcert> in case it is an encrypted PKCS#12 file.
+If not given here, the password will be prompted for if needed.
+
+For more information about the format of B<arg> see the
+B<PASS PHRASE ARGUMENTS> section in L<openssl(1)>.
+
+=item B<-digest> I<name>
+
+Specifies name of supported digest to use in RFC 4210's MSG_SIG_ALG
+and as the one-way function (OWF) in MSG_MAC_ALG.
+If applicable, this is used for message protection and
+Proof-of-Possession (POPO) signatures.
+To see the list of supported digests, use B<openssl list -digest-commands>.
+Defaults to C<sha256>.
+
+=item B<-mac> I<name>
+
+Specifies name of supported digest to use as the MAC algorithm in MSG_MAC_ALG.
```

Comment:
This sentence makes no sense. A digest is not a MAC algorithm.

Reply:
changed